### DISCUSSION TITLE
Intellisense showing error when variable length arrays are declared, even though the compiler supports them.

### DISCUSSION THREAD

**Discussion by SciSaif** [microsoft/vscode-cpptools#7986] [Q&A]
https://github.com/microsoft/vscode-cpptools/issues/7593#issue-901655681

i have asked this problem before. 
i started having the same issue again. 
but this time if i do something like 
const int n = 5;
int arr[n];
it works
but if i do 
int n = 5;
int arr[n]; 
it shows the same error

@sean-mcmanus

**Comment by sean-mcmanus** [comment]
It's working for me. Can you run C/C++: Log Diagnostics and see what it reports for the IntelliSense Mode?

**Reply by SciSaif** [reply]
Here's the log 

> loggingLevel has changed to: Debug
File exclude: **/.vscode
File exclude: **/.git
File exclude: **/.svn
File exclude: **/.hg
File exclude: **/CVS
File exclude: **/.DS_Store
File exclude: **/.classpath
File exclude: **/.project
File exclude: **/.settings
File exclude: **/.factorypath
Search exclude: **/node_modules
Search exclude: **/bower_components
Search exclude: **/*.code-search
Discovering files...
  Processing folder (non-recursive): C:/PROGRAM FILES (X86)/MICROSOFT VISUAL STUDIO/2019/COMMUNITY/VC/TOOLS/MSVC/14.28.29333/INCLUDE
  Processing folder (recursive): C:/PROGRAM FILES (X86)/WINDOWS KITS/10/INCLUDE/10.0.17763.0/CPPWINRT/
  Processing folder (recursive): C:/PROGRAM FILES (X86)/WINDOWS KITS/10/INCLUDE/10.0.17763.0/SHARED/
  Processing folder (recursive): C:/PROGRAM FILES (X86)/WINDOWS KITS/10/INCLUDE/10.0.17763.0/UCRT/
  Processing folder (recursive): C:/PROGRAM FILES (X86)/WINDOWS KITS/10/INCLUDE/10.0.17763.0/UM/
  Processing folder (recursive): C:/PROGRAM FILES (X86)/WINDOWS KITS/10/INCLUDE/10.0.17763.0/WINRT/
  Processing folder (recursive): F:/MEGA/C++ COURSE/
  Discovering files: 5030 file(s) processed
  4 file(s) removed from database
Done discovering files.
Populate include completion cache.
Parsing remaining files...
  Parsing: 1 files(s) processed
Done parsing remaining files.
  tag parsing file: F:\MEGA\C++ COURSE\TOPICS\ROUGH.CPP
  tag parsing file: C:\PROGRAM FILES (X86)\MICROSOFT VISUAL STUDIO\2019\COMMUNITY\VC\TOOLS\MSVC\14.28.29333\INCLUDE\BITS\STDC++.H
  tag parsing file: C:\PROGRAM FILES (X86)\MICROSOFT VISUAL STUDIO\2019\COMMUNITY\VC\TOOLS\MSVC\14.28.29333\INCLUDE\EXPERIMENTAL\RESUMABLE
  tag parsing file: C:\PROGRAM FILES (X86)\MICROSOFT VISUAL STUDIO\2019\COMMUNITY\VC\TOOLS\MSVC\14.28.29333\INCLUDE\EXPERIMENTAL\COROUTINE
  tag parsing file: C:\PROGRAM FILES (X86)\MICROSOFT VISUAL STUDIO\2019\COMMUNITY\VC\TOOLS\MSVC\14.28.29333\INCLUDE\CODEANALYSIS\SOURCEANNOTATIONS.H
idle loop: reparsing the active document
Checking for syntax errors: file:///f%3A/Mega/C%2B%2B%20Course/topics/rough.cpp
Queueing IntelliSense update for files in translation unit of: F:\MEGA\C++ COURSE\TOPICS\ROUGH.CPP
Error squiggle count: 0
Update IntelliSense time (sec): 0.254
Offering completion
sending 1 changes to server
sending 1 changes to server
idle loop: reparsing the active document
Checking for syntax errors: file:///f%3A/Mega/C%2B%2B%20Course/topics/rough.cpp
Queueing IntelliSense update for files in translation unit of: F:\MEGA\C++ COURSE\TOPICS\ROUGH.CPP
sending 1 changes to server
sending 1 changes to server
sending 1 changes to server
sending 1 changes to server
sending 1 changes to server
sending 1 changes to server
sending 1 changes to server
sending 1 changes to server
sending 1 changes to server
sending 1 changes to server
sending 1 changes to server
sending 1 changes to server
Error squiggle count: 0
Checking for syntax errors: file:///f%3A/Mega/C%2B%2B%20Course/topics/rough.cpp
Queueing IntelliSense update for files in translation unit of: F:\MEGA\C++ COURSE\TOPICS\ROUGH.CPP
  tag parsing file: F:\MEGA\C++ COURSE\TOPICS\ROUGH.CPP
Error squiggle count: 0
Update IntelliSense time (sec): 9.978
Offering completion
sending 1 changes to server
Offering completion
sending 1 changes to server
Checking for syntax errors: file:///f%3A/Mega/C%2B%2B%20Course/topics/rough.cpp
Queueing IntelliSense update for files in translation unit of: F:\MEGA\C++ COURSE\TOPICS\ROUGH.CPP
  tag parsing file: F:\MEGA\C++ COURSE\TOPICS\ROUGH.CPP
sending 1 changes to server
sending 1 changes to server
sending 1 changes to server
Error squiggle count: 1
Offering completion
sending 1 changes to server
sending 1 changes to server
sending 1 changes to server
sending 1 changes to server
Checking for syntax errors: file:///f%3A/Mega/C%2B%2B%20Course/topics/rough.cpp
Queueing IntelliSense update for files in translation unit of: F:\MEGA\C++ COURSE\TOPICS\ROUGH.CPP
  tag parsing file: F:\MEGA\C++ COURSE\TOPICS\ROUGH.CPP
idle loop: reparsing the active document
sending 1 changes to server
sending 1 changes to server
Checking for syntax errors: file:///f%3A/Mega/C%2B%2B%20Course/topics/rough.cpp
Queueing IntelliSense update for files in translation unit of: F:\MEGA\C++ COURSE\TOPICS\ROUGH.CPP
Error squiggle count: 1
Update IntelliSense time (sec): 7.779

**Reply by sean-mcmanus** [reply]
Those aren't the logs from C/C++: Log Diagnostics, but it appears that it's incorrectly using msvc mode instead of gcc mode. Do you have your intelliSenseMode and compilerPath set correctly?

**Reply by SciSaif** [reply]
![image](https://user-images.githubusercontent.com/51053279/130028045-b51ba27e-85ca-4acd-8353-29414af05eb8.png)
here
I don't know how to check the log diagnostics :(

**Reply by sean-mcmanus** [reply]
From the main menu, choose View->Command Palette and select the "C/C++: Log Diagnostics" command after opening a file. The output window should show the diagnostics pane output.

You're setting the default IntelliSense mode, but the default might be getting overridden by a c_cpp_properties.json or some other setting.

Also, what version of the extension are you using?

**Reply by SciSaif** [reply]
log: 

> -------- Diagnostics - 8/20/2021, 3:04:44 AM
Version: 1.6.0-insiders2
Current Configuration:
{
    "name": "Win32",
    "includePath": [
        "${workspaceFolder}/**"
    ],
    "defines": [
        "_DEBUG",
        "UNICODE",
        "_UNICODE"
    ],
    "windowsSdkVersion": "10.0.17763.0",
    "compilerPath": "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community\\VC\\Tools\\MSVC\\14.28.29333\\bin\\Hostx64\\x64\\cl.exe",
    "compilerArgs": [],
    "cStandard": "c17",
    "cppStandard": "c++20",
    "intelliSenseMode": "windows-gcc-x86",
    "intelliSenseModeIsExplicit": true,
    "cStandardIsExplicit": true,
    "cppStandardIsExplicit": true,
    "compilerPathIsExplicit": false,
    "browse": {
        "path": [
            "${workspaceFolder}/**"
        ],
        "limitSymbolsToIncludedHeaders": true
    }
}
Translation Unit Mappings:
[ F:\Mega\C++ Course\dsa course\DynamicProgramming\maximumCuts.cpp ]:
    F:\MEGA\C++ COURSE\DSA COURSE\DYNAMICPROGRAMMING\MAXIMUMCUTS.CPP
Translation Unit Configurations:
[ F:\Mega\C++ Course\dsa course\DynamicProgramming\maximumCuts.cpp ]:
    Process ID: 13528
    Memory Usage: 127 MB
    Compiler Path: C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29333\bin\Hostx64\x64\cl.exe
    Includes:
        C:\PROGRAM FILES (X86)\MICROSOFT VISUAL STUDIO\2019\COMMUNITY\VC\TOOLS\MSVC\14.28.29333\INCLUDE
        C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.17763.0\UM
        C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.17763.0\UCRT
        C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.17763.0\SHARED
        C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.17763.0\WINRT
        C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.17763.0\CPPWINRT
    Defines:
        _DEBUG
        UNICODE
        _UNICODE
    Standard Version: ms_c++latest
    IntelliSense Mode: windows-msvc-x86
Total Memory Usage: 127 MB

> ------- Workspace parsing diagnostics -------
> Number of files discovered (not excluded): 5030
> Number of files parsed: 494



version: v1.6.0-insiders3

Launch.json file (don't know if this is needed)
`{
    // Use IntelliSense to learn about possible attributes.
    // Hover to view descriptions of existing attributes.
    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
    "version": "0.2.0",
    "configurations": [

        {
            "name": "g++.exe - Build and debug active file",
            "type": "cppdbg",
            "request": "launch",
            "program": "${fileDirname}\\${fileBasenameNoExtension}.exe",
            "args": [],
            "stopAtEntry": false,
            "cwd": "${workspaceFolder}",
            "environment": [],
            "externalConsole": false,
            "MIMode": "gdb",
            "miDebuggerPath": "gdb",
            "setupCommands": [
                {
                    "description": "Enable pretty-printing for gdb",
                    "text": "-enable-pretty-printing",
                    "ignoreFailures": true
                }
            ],
            "preLaunchTask": "C/C++: g++.exe build active file"
        }
    ]
}`

**Reply by sean-mcmanus** [reply]
You need to delete `"compilerPath": "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29333\bin\Hostx64\x64\cl.exe",` -- that value is causing windows-msvc-x86 to be used as the intelliSenseMode.

**Reply by SciSaif** [reply]
where do i delete it from?
![image](https://user-images.githubusercontent.com/51053279/130149930-fed710b8-ad0e-49ad-b4a2-b868d899025b.png)
should i delete this file?

**Reply by sean-mcmanus** [reply]
Don't delete the file, just the setting In your c_cpp_properties.json or your C_Cpp.default.compilerPath in a settings json file.

**Reply by SciSaif** [reply]
settings.json
```
{
    "editor.suggestSelection": "first",
    "vsintellicode.modify.editor.suggestSelection": "automaticallyOverrodeDefaultValue",
    "editor.mouseWheelZoom": true,
    "code-runner.clearPreviousOutput": true,
    "code-runner.runInTerminal": true,
    "explorer.confirmDelete": false,
    "C_Cpp.updateChannel": "Insiders",
    "workbench.iconTheme": "eq-material-theme-icons-palenight",
    "workbench.preferredHighContrastColorTheme": "Community Material Theme Palenight High Contrast",
    "workbench.colorTheme": "Material Theme Ocean",
    "C_Cpp.default.cppStandard": "c++20",
    "C_Cpp.default.cStandard": "c17",
    "files.exclude": {
        "**/.classpath": true,
        "**/.project": true,
        "**/.settings": true,
        "**/.factorypath": true
    },
    "workbench.editorAssociations": {
        "*.ipynb": "jupyter.notebook.ipynb"
    },
    "files.associations": {
        "**/*.py": "python"
    },
    "cph.general.defaultLanguage": "cpp",
    "C_Cpp.experimentalFeatures": "Enabled",
    "cph.general.defaultLanguageTemplateFileLocation": "F:\\Mega\\C++ Course\\useful\\cpTemplate.cpp",
    "[cpp]": {
        "editor.defaultFormatter": "ms-vscode.cpptools"
    },
    "C_Cpp.clang_format_fallbackStyle": "{ BasedOnStyle: Google, IndentWidth: 4, ColumnLimit: 0}",
    "C_Cpp.default.intelliSenseMode": "windows-gcc-x86",
    "cph.general.autoShowJudge": false

}


```

is this the file you are talking about?
where is c_cpp_properties.json located?

Also these are the only files in .vscode
![image](https://user-images.githubusercontent.com/51053279/130152192-84241285-5b0a-47fb-8df0-6dd49d1870e1.png)
settings.json in the above picture
```
{
    "python.pythonPath": "C:\\Users\\user\\AppData\\Local\\Programs\\Python\\Python39\\python.exe",
    "files.associations": {
        "**/*.py": "python",
        "vector": "cpp",
        "unordered_map": "cpp",
        "deque": "cpp",
        "list": "cpp",
        "xstring": "cpp",
        "iostream": "cpp",
        "utility": "cpp",
        "xtree": "cpp",
        "xtr1common": "cpp",
        "xhash": "cpp",
        "unordered_set": "cpp",
        "tuple": "cpp",
        "map": "cpp",
        "xiosbase": "cpp",
        "cmath": "cpp",
        "ostream": "cpp",
        "queue": "cpp",
        "xutility": "cpp",
        "stack": "cpp",
        "algorithm": "cpp",
        "chrono": "cpp",
        "iterator": "cpp",
        "new": "cpp",
        "complex": "cpp",
        "array": "cpp",
        "atomic": "cpp",
        "*.tcc": "cpp",
        "bitset": "cpp",
        "cctype": "cpp",
        "cfenv": "cpp",
        "charconv": "cpp",
        "cinttypes": "cpp",
        "clocale": "cpp",
        "codecvt": "cpp",
        "condition_variable": "cpp",
        "csetjmp": "cpp",
        "csignal": "cpp",
        "cstdarg": "cpp",
        "cstddef": "cpp",
        "cstdint": "cpp",
        "cstdio": "cpp",
        "cstdlib": "cpp",
        "cstring": "cpp",
        "ctime": "cpp",
        "cuchar": "cpp",
        "cwchar": "cpp",
        "cwctype": "cpp",
        "forward_list": "cpp",
        "exception": "cpp",
        "functional": "cpp",
        "memory": "cpp",
        "memory_resource": "cpp",
        "numeric": "cpp",
        "optional": "cpp",
        "random": "cpp",
        "ratio": "cpp",
        "regex": "cpp",
        "set": "cpp",
        "string": "cpp",
        "string_view": "cpp",
        "system_error": "cpp",
        "type_traits": "cpp",
        "fstream": "cpp",
        "future": "cpp",
        "initializer_list": "cpp",
        "iomanip": "cpp",
        "iosfwd": "cpp",
        "istream": "cpp",
        "limits": "cpp",
        "mutex": "cpp",
        "scoped_allocator": "cpp",
        "shared_mutex": "cpp",
        "sstream": "cpp",
        "stdexcept": "cpp",
        "streambuf": "cpp",
        "thread": "cpp",
        "typeindex": "cpp",
        "typeinfo": "cpp",
        "valarray": "cpp"
    },
    "C_Cpp.loggingLevel": "Debug"
}
```

**Reply by sean-mcmanus** [reply]
Are you compiling with a gcc compiler still, like mingw? You need to set your C_Cpp.default.compilerPath to the path of your compiler. We don't appear to be auto-detecting it for some reason (is it not installed at the default location?). Or do you want to compile with cl.exe?

**Reply by SciSaif** [reply]
yes it is not installed at the default location i guess.
i will just install mingw again.
